### ISSUE TITLE
.iex.exs error behaviour

### ISSUE THREAD

**Issue by ConnorRigby** [elixir-lang/elixir#8618]
### Environment

* Elixir & Erlang/OTP versions (elixir --version):  Elixir 1.7.3 (compiled with Erlang/OTP 21)
* Operating system:  Linux

### Current behavior

Loading `.iex.exs` with compile or runtime errors currently halts the system. This is currently intentional as seen [here](https://github.com/elixir-lang/elixir/blob/d1e3e4dafd48e55898ec62235d6f91ab5aa7bc64/lib/iex/lib/iex/evaluator.ex#L182-L197). 

### Expected behavior

I'd like to discuss if this is the best behaviour. For Nerves (and i suspect any other Distillyery based release system) this will cause the vm boot to fail. Would there be interest in changing this to either just _not_ halt the system, or have the behaviour configurable via the command line or something?

**Comment by josevalim** [comment]
@ConnorRigby agreed. Can you please send a PR that removes the halting and have only the logging and add a test as well?
